plutus-ledger-constraints/src/Ledger/Constraints/OnChain/V2.hs

```diff
@@ -67,26 +68,35 @@ checkOwnOutputConstraint
     -> ScriptOutputConstraint o
     -> Bool
 checkOwnOutputConstraint ctx@ScriptContext{scriptContextTxInfo} ScriptOutputConstraint{ocDatum, ocValue} =
```

Comment:
Shouldn't we (maybe in another ticket) factorize the common behaviour of Onchain.V1 and OnChain.V2 (maybe by reusing in V2 the verification parts that are still valid from V1, like in plutus-api)? The duplication here looks error-prone (this PR doesn't introduce the problem but highlights it).